library/src/com/google/maps/android/PolyUtil.java

```diff
@@ -183,15 +211,17 @@ private static boolean isLocationOnEdgeOrPath(LatLng point, List<LatLng> poly, b
         LatLng prev = poly.get(closed ? size - 1 : 0);
         double lat1 = toRadians(prev.latitude);
         double lng1 = toRadians(prev.longitude);
+        int idx = 0;
         if (geodesic) {
             for (LatLng point2 : poly) {
                 double lat2 = toRadians(point2.latitude);
                 double lng2 = toRadians(point2.longitude);
                 if (isOnSegmentGC(lat1, lng1, lat2, lng2, lat3, lng3, havTolerance)) {
-                    return true;
+                    return idx;
```

Comment:
It may be useful to clearly state the meaning of a return of 0 (idx==0), notably for closed==true. It may be more intuitive to map the "closing" segment on idx==n-1 instead of idx==0 as is now.